### DISCUSSION TITLE
Hierarchical linear regression by applying (multiple) linear regression multiple times?

### DISCUSSION THREAD

**Discussion by teanijarv** [raphaelvallat/pingouin#339] [General]
May be a dumb question, but can I perform hierarchical linear regression by doing each steps separetely with `pingouin.linear_regression` function?

For example:
```
lm_step1 = pg.linear_regression(df['sex', df['tip'])
lm_step2 = pg.linear_regression(df[['total_bill', 'size']], df['tip'])
```

What should be the next steps - how could I compare these two regressions etc?

Pingouin is an amazing package for someone still learning stats and it's really well documented. Great job! I'm just learning about how to do hierarchical linear regression and I really don't want to use SPSS..

**Comment by raphaelvallat** [comment]
Hi @teanijarv! Thanks for the kind words. You can compare the R^2 of the different models to get a rough idea of the increase in model fit. But Pingouin does not implement the tool/function for a more formal model comparison. I would recommend using R, if you're familiar with it.
